Design of stream pages

Flaburgan opened the pull request:
I didn't like the dark grey background we currently have on the stream pages with a wide screen. I have several ideas I want to try about what the stream pages can become. They can be controversial though so I wanted to first open this PR with changes which could be accepted by everybody.

**Reminder: the current stream view looks like that on the develop branch:**
![current-stream](https://cloud.githubusercontent.com/assets/930064/12215457/46b79c82-b6bd-11e5-9d9b-b06a123ad427.png)

This pull request removes the right side bar, moving the content in the left side bar. That way, 80% of the screen is available to the stream. It also moves the user avatar and name at the left of the publisher, which helps users to know with which account they are connected and avoid a too big width for the publisher.

Here is how it looked (at the beginning of this PR, this is an old screenshot, kept for history, see the up-to-date screenshot at the end of this post) for a 1440px width screen:

![capture d ecran 2015-11-07 a 16 45 46](https://cloud.githubusercontent.com/assets/930064/11015889/c2746816-856f-11e5-9528-e5cbb8a1010e.png)

I took care of mobile in this PR even if this is the desktop version. I know @AugierLe42e  will be happy about that ;) The content of what was the right side bar disappears on small screen to directly access the stream and not have to scroll down a lot. The avatar and the name next to the publisher also disappear on xs and sm screens.

**Here is how the stream currently looks with this PR:**
![](https://cloud.githubusercontent.com/assets/930064/11194627/637eb3a8-8cad-11e5-9c99-98aa0ebc7028.png)

Flaburgan committed:
collapse the sidebars to the side of the screen even on big screens

Drop right side bar, move content to left side bar

Move user avatar and name from the left nav bar to the left of the publisher

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
-      #aspect_stream_container.stream_container
-        = render 'aspects/aspect_stream', stream: @stream
+      .info-bar.hidden-xs
+        #selected_aspect_contacts.section
```
Classes should be listed before IDs (.section should precede #selected_aspect_contacts)

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+          .section
+            .title
+              %h5.title-header
+                = t('shared.invitations.invite_your_friends')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+        .section
+          .title
+            %h5.title-header
+              = t('aspects.index.new_here.title')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+            %h5.title-header
+              = t('aspects.index.new_here.title')
+          .content
+            != t('aspects.index.new_here.follow', link: link_to("#"+t('shared.publisher.new_user_prefill.newhere'), tag_path(name: t('shared.publisher.new_user_prefill.newhere'))))
```
Line is too long. [180/120]

diaspora-code-review replied:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review replied:
Surrounding space missing for operator `+`.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+          .content
+            != t('aspects.index.new_here.follow', link: link_to("#"+t('shared.publisher.new_user_prefill.newhere'), tag_path(name: t('shared.publisher.new_user_prefill.newhere'))))
+            %br
+            = link_to(t('aspects.index.new_here.learn_more'), "http://wiki.diasporafoundation.org/Welcoming_Committee")
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+        .section
+          .title
+            %h5.title-header
+              = t('aspects.index.help.need_help')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              = t('aspects.index.help.need_help')
+          .content
+            %p
+              = t('aspects.index.help.here_to_help')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+            %p
+              = t('aspects.index.help.here_to_help')
+            %p
+              = t('aspects.index.help.do_you')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              = t('aspects.index.help.do_you')
+            %ul
+              %li
+                != t('aspects.index.help.have_a_question', :link => link_to("#"+t('aspects.index.help.tag_question'), tag_path(:name => t('aspects.index.help.tag_question'))))
```
Line is too long. [175/120]

diaspora-code-review replied:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review replied:
Use the new Ruby 1.9 hash syntax.

diaspora-code-review replied:
Surrounding space missing for operator `+`.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              %li
+                != t('aspects.index.help.have_a_question', :link => link_to("#"+t('aspects.index.help.tag_question'), tag_path(:name => t('aspects.index.help.tag_question'))))
+              %li
+                != t('aspects.index.help.find_a_bug', :link => link_to("#"+t('aspects.index.help.tag_bug'), tag_path(:name => t('aspects.index.help.tag_bug'))))
```
Line is too long. [160/120]

diaspora-code-review replied:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review replied:
Use the new Ruby 1.9 hash syntax.

diaspora-code-review replied:
Surrounding space missing for operator `+`.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              %li
+                != t('aspects.index.help.find_a_bug', :link => link_to("#"+t('aspects.index.help.tag_bug'), tag_path(:name => t('aspects.index.help.tag_bug'))))
+              %li
+                != t('aspects.index.help.feature_suggestion', :link => link_to("#"+t('aspects.index.help.tag_feature'), tag_path(:name => t('aspects.index.help.tag_feature'))))
```
Line is too long. [176/120]

diaspora-code-review replied:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review replied:
Use the new Ruby 1.9 hash syntax.

diaspora-code-review replied:
Surrounding space missing for operator `+`.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              %li
+                != t('aspects.index.help.feature_suggestion', :link => link_to("#"+t('aspects.index.help.tag_feature'), tag_path(:name => t('aspects.index.help.tag_feature'))))
+            %p
+              != t('aspects.index.help.tutorials_and_wiki',
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review replied:
Use the new Ruby 1.9 hash syntax.

diaspora-code-review replied:
Space missing after comma.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+            %p
+              != t('aspects.index.help.tutorials_and_wiki',
+                   :faq => link_to(t('_help'), help_path),
+                   :tutorial => link_to(t('aspects.index.help.tutorial_link_text'), "https://diasporafoundation.org/tutorials", :target => '_blank'),
```
Line is too long. [149/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              != t('aspects.index.help.tutorials_and_wiki',
+                   :faq => link_to(t('_help'), help_path),
+                   :tutorial => link_to(t('aspects.index.help.tutorial_link_text'), "https://diasporafoundation.org/tutorials", :target => '_blank'),
+                   :wiki => link_to('Wiki','http://wiki.diasporafoundation.org', :target => '_blank'), :target => '_blank')
```
Line is too long. [123/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+          .section
+            .title
+              %h5.title-header
+                = t('aspects.index.services.heading')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+                = t('aspects.index.services.heading')
+            .content
+              %div
+                = t('aspects.index.services.content')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              #right_service_icons
+                - AppConfig.configured_services.each do |service|
+                  - if AppConfig.show_service?(service, current_user)
+                    - unless current_user.services.any?{|x| x.provider == service}
```
Space missing to the left of {.

diaspora-code-review replied:
Space missing inside }.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+                - AppConfig.configured_services.each do |service|
+                  - if AppConfig.show_service?(service, current_user)
+                    - unless current_user.services.any?{|x| x.provider == service}
+                      = link_to(content_tag(:div, nil, :class => "social_media_logos-#{service.to_s.downcase}-24x24", :title => service.to_s.titleize), "/auth/#{service}")
```
Line is too long. [171/120]

diaspora-code-review replied:
Use the new Ruby 1.9 hash syntax.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+        .section
+          .title
+            %h5.title-header
+              = t('bookmarklet.heading')
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+            %h5.title-header
+              = t('bookmarklet.heading')
+          .content
+            != t('bookmarklet.explanation', :link => link_to(t('bookmarklet.post_something'), bookmarklet_code))
```
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

diaspora-code-review replied:
Use the new Ruby 1.9 hash syntax.

Flaburgan commented:
Wow, I ran `bin/pronto run --index` before committing and I corrected every errors. What's the command to run it when changes are already commited? Looks like this was my mistake here.

jhass commented:
like in the wiki, `-c develop`

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+            %h5.title-header
+              = t("aspects.index.new_here.title")
+          .content
+            != t("aspects.index.new_here.follow", link: link_to("#" + t("shared.publisher.new_user_prefill.newhere"), tag_path(name: t("shared.publisher.new_user_prefill.newhere"))))
```
Line is too long. [182/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              = t("aspects.index.help.do_you")
+            %ul
+              %li
+                != t("aspects.index.help.have_a_question", link: link_to("#" + t("aspects.index.help.tag_question"), tag_path(name: t("aspects.index.help.tag_question"))))
```
Line is too long. [171/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              %li
+                != t("aspects.index.help.have_a_question", link: link_to("#" + t("aspects.index.help.tag_question"), tag_path(name: t("aspects.index.help.tag_question"))))
+              %li
+                != t("aspects.index.help.find_a_bug", link: link_to("#" + t("aspects.index.help.tag_bug"), tag_path(name: t("aspects.index.help.tag_bug"))))
```
Line is too long. [156/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              %li
+                != t("aspects.index.help.find_a_bug", link: link_to("#" + t("aspects.index.help.tag_bug"), tag_path(name: t("aspects.index.help.tag_bug"))))
+              %li
+                != t("aspects.index.help.feature_suggestion", link: link_to("#" + t("aspects.index.help.tag_feature"), tag_path(name: t("aspects.index.help.tag_feature"))))
```
Line is too long. [172/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+            %p
+              != t("aspects.index.help.tutorials_and_wiki",
+                   faq: link_to(t("_help"), help_path),
+                   tutorial: link_to(t("aspects.index.help.tutorial_link_text"), "https://diasporafoundation.org/tutorials", target: "_blank"),
```
Line is too long. [143/120]

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+              .right-service-icons
+                - AppConfig.configured_services.each do |service|
+                  - if AppConfig.show_service?(service, current_user)
+                    - unless current_user.services.any?{|x| x.provider == service}
```
Space missing to the left of {.

diaspora-code-review replied:
Space missing inside }.

diaspora-code-review commented on app/views/streams/main_stream.html.haml:
```diff
@@ -44,20 +38,104 @@
+                - AppConfig.configured_services.each do |service|
+                  - if AppConfig.show_service?(service, current_user)
+                    - unless current_user.services.any?{|x| x.provider == service}
+                      = link_to(content_tag(:div, nil, class: "social_media_logos-#{service.to_s.downcase}-24x24", title: service.to_s.titleize), "/auth/#{service}")
```
Line is too long. [165/120]

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -51,7 +51,8 @@
   }
 
-  #aspects_list, #tags_list {
+  #aspects_list,
```
Avoid using id selectors

diaspora-code-review replied:
Selector `aspects_list` should be written in lowercase with hyphens

diaspora-code-review replied:
Selector `tags_list` should be written in lowercase with hyphens

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+          padding-left: 0;
+        }
+
+        > #invite_code {
```
Avoid using id selectors

diaspora-code-review replied:
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review replied:
Selector `invite_code` should be written in lowercase with hyphens

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+    .section {
+      margin-bottom: 20px;
+
+      > .title {
```
Nesting should be no greater than 3, but was 4

diaspora-code-review replied:
Selector should have depth of applicability no greater than 3, but was 4

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+        }
+      }
+
+      .content {
```
Nesting should be no greater than 3, but was 4

diaspora-code-review replied:
Selector should have depth of applicability no greater than 3, but was 4

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+        border-bottom: 1px solid $border-grey;
+        padding-bottom: 5px;
+
+        h5 {
```
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+        line-height: 18px;
+        padding: 10px 0;
+
+        p,
```
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+          margin: 0;
+        }
+
+        ul {
```
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+        ul {
+          margin-bottom: 5px;
+          padding-left: 0;
+          li { list-style: none; }
```
Selector should have depth of applicability no greater than 3, but was 6

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+          li { list-style: none; }
+        }
+
+        .btn-link {
```
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+          width: 100%;
+        }
+
+        > .right-service-icons {
```
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+          }
+          text-align: center;
+
+          .social_media_logos-facebook-24x24,
```
Selector should have depth of applicability no greater than 3, but was 6

diaspora-code-review replied:
Selector `social_media_logos-facebook-24x24` should be written in lowercase with hyphens

diaspora-code-review replied:
Selector `social_media_logos-twitter-24x24` should be written in lowercase with hyphens

diaspora-code-review replied:
Selector `social_media_logos-tumblr-24x24` should be written in lowercase with hyphens

diaspora-code-review replied:
Selector `social_media_logos-wordpress-24x24` should be written in lowercase with hyphens

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+            width: 24px;
+          }
+
+          a {
```
Selector should have depth of applicability no greater than 3, but was 6

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+          }
+        }
+
+        .stream-faces {
```
Selector should have depth of applicability no greater than 3, but was 5

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +112,78 @@
+        .stream-faces {
+          display: flex;
+          flex-flow: row wrap;
+          img { margin: 1px; }
```
Selector should have depth of applicability no greater than 3, but was 6

diaspora-code-review commented on app/views/publisher/_publisher.mobile.haml:
```diff
@@ -11,7 +11,7 @@
       - if current_user.services
         - for service in current_user.services
-          = image_tag "social_media_logos/#{service.provider}-32x32.png", title: service.provider.titleize, class: "service_icon dim", id:"#{service.provider}", maxchar: "#{service.class::MAX_CHARACTERS}"
+          = image_tag "social-media-logos/#{service.provider}-32x32.png", title: service.provider.titleize, class: "service_icon dim", id:"#{service.provider}", maxchar: "#{service.class::MAX_CHARACTERS}"
```
Line is too long. [204/120]

diaspora-code-review replied:
Space missing after colon.

diaspora-code-review commented on app/assets/stylesheets/publisher.scss:
```diff
@@ -41,7 +41,7 @@
         .btn.btn-link.question_mark:hover .entypo-cog { color: $black; }
         .dim { opacity: 0.3; }
-        .social_media_logos-wordpress-16x16 {
+        .social-media-logos-wordpress-16x16 {
```
Selector should have depth of applicability no greater than 3, but was 5

Flaburgan commented:
I corrected most of the pronto errors. Depth of selectors will be very time consuming to fix, is it a strong requirement just a nice to have?

kas70 commented:
Hi,
i don't know if this is the right point to discuss this idea here, but i think it would be cool, if the left sidebar (stream, my activity, mentions,...) stays in place if you scroll down your stream.
![stream2](https://cloud.githubusercontent.com/assets/922628/11017339/49220b3a-859c-11e5-9a4f-1702db7ad33a.png)

AugierLe42e commented:
Yep, I proposed the same thing to @Flaburgan

Globulle commented:
I like the effort to better use the full width of the screen. However we have to keep in mind that long lines are not easily readable. Having wide posts is not a problem when they contain one or two sentences, but as soon as they become text blocks (a frequent situation on D*), [the length of the line matters](http://baymard.com/blog/line-length-readability) for a comfortable reading.

AugierLe42e commented:
@Globulle : You're right. I'd like to temperate a bit, though. Firstly, this is still a WIP. I don't doubt that a discussion on the size of the elements of the page is going to start soon. Secondly, @Flaburgan said on IRC that the goal of this layout was to let the chat be always displayed which I found to be a good idea. Thirdly, this new layout is on a good track to bring some consistency o all pages. The stream was the only page with two sidebars which is causing us hard time to get a uniform styling with the rest of the website.

On the other hand, your article states that 50-60 chars is optimal. I don't really agree on that. Especially with a full HD screen, having a tiny main column on a large layout is very annoying. To compare with the modern coding styles, line lenght of 120 chars on a line are now a standard on all IDEs. I think some other efforts can be made in other domains rather than sacrify space. For instance, font-size, which is now 13px can be increased to 14px. Posts layout can be improved to increase content weight and the publisher can be reworked to easier preview before posting.

Globulle commented:
@AugierLe42e I just wanted to point it out to be sure we have this in mind. I'm glad you confirm this is the case. :-)

I agree 50-60 chars seems very short. By the way, this is also the default setting of LaTeX, [as explained here](http://tex.stackexchange.com/a/71211/46718). Despite I remain convinced it makes a big difference, let's not argue about that. I also agree there are others parameters that can be taken into account.
Maybe two columns of posts may also be a good solution (I know @Flaburgan has already thought about that)? Maybe one for followed tags and the other one for aspects' posts? This is just an idea.

Flaburgan commented:
Still WIP. Things to do:
- [ ] Fix tests
- [ ] Fix small regression on profile page (no more margin between the publisher and the first post)
- [ ] Fix getting started pophover which are now not displayed at the correct places
- [ ] Also remove the dark side left margin on tags page
- [ ] Make username and publisher to look better
- [ ] [Optional ?] Fix pronto errors

Flaburgan commented:
@Globulle I completely agree a text with more than 120 cols is hard to read. As answered by @AugierLe42e, we want a one column layout and I think we could always display the chat there. I also experiment on a possible two columns layout on [this branch](https://github.com/Flaburgan/diaspora/commits/masonry). This would represent a bigger change though and everybody may not like it, that's why I opened this pull request first.

If you have other suggestions about how to use this free space, please do! (Oh, apart for "put ads there" of course. This is a facebook thing :p)

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -49,9 +50,11 @@
   }
 
-  #aspects_list, #tags_list {
+  #aspects_list,
```
Avoid using id selectors

diaspora-code-review replied:
Selector `aspects_list` should be written in lowercase with hyphens

diaspora-code-review replied:
Selector `tags_list` should be written in lowercase with hyphens

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -135,4 +114,78 @@
+          }
+          text-align: center;
+
+          .social-media-logos-facebook-24x24,
```
Selector should have depth of applicability no greater than 3, but was 6

diaspora-code-review commented on app/assets/stylesheets/stream.scss:
```diff
@@ -1,10 +1,25 @@
 }
 
 .stream_container {
+  .well#ignore-info { text-align: center; }
```
Avoid using id selectors

diaspora-code-review commented on app/assets/stylesheets/stream.scss:
```diff
@@ -1,10 +1,25 @@
+
   #publisher {
-    margin-bottom: 15px;
+    margin-left: 65px;
```
Properties should be ordered margin-bottom, margin-left

Flaburgan committed:
remove dark gray left column on tag page

ceathar commented:
Thanks for all that work, and good idea. I just would like to add that apart from the text, the pictures (and the videos) become also too large (in height) to be seen entirely in the stream.

Flaburgan committed:
Fix getting started help

Add temp max-width to the streams

Back to white bg for left menu

Fix invitation code selection

fix pronto errors

diaspora-code-review commented on app/assets/stylesheets/getting-started.scss:
```diff
@@ -42,13 +42,16 @@
   .close { line-height: 18px; }
 }
 
+#gs-shim {
```
Avoid using id selectors

diaspora-code-review commented on app/assets/stylesheets/getting-started.scss:
```diff
@@ -42,13 +42,16 @@
   box-shadow: inset 0 -2px 10px rgba(0,0,0,0.35);
-  margin-bottom: 20px;
-  margin-top: -40px;
+  padding-top: 20px;
```
Properties should be ordered background, box-shadow, padding-bottom, padding-top

Flaburgan commented:
Here is how this PR actually looks like:
![streams-page](https://cloud.githubusercontent.com/assets/930064/11194617/5848e512-8cad-11e5-9f16-ecc8a6314cc2.png)
![open-streams](https://cloud.githubusercontent.com/assets/930064/11194627/637eb3a8-8cad-11e5-9c99-98aa0ebc7028.png)
![tags-page](https://cloud.githubusercontent.com/assets/930064/11194616/584871ea-8cad-11e5-9be7-c068b89025d7.png)

We're back to a left menu with a white background on the streams and tag pages. The grey text on grey background looked difficult to read to me (contrast was insufficient) + I wanted to have the same colors everywhere. The streams page was the only one with a darker grey as background. Now, streams, tag, people (including photos and contacts tabs) and conversation pages have the same grey background, and white menu on the left.

@svbergerem I would love to hear your opinion about my work. It's not easy to see how it looks on screenshots, tell me if you still have access to your account on d-fr.

Globulle commented:
I like it, @Flaburgan . I have a suggestion: on the stream view (first image), I think it would be good to differentiate more clearly the part where interaction is possible and the part which is just informative (basically what was in the left and in the right in previous version, respectively). The menus Flux/Mon activité/... don't look like as clickable items to me.

One way to solve this would be to add small icons to these labels (something I have in mind for months, maybe such an issue already exist). We could use an @ sign for mentions and a # for tags, which would allow to remove them from the current labels. This would be cleaner.

AugierLe42e commented:
> The streams page was the only one with a darker grey as background.

Nope, tag page has too.

This new presentation has a very stange look on full HD screens. It leaves a lots of space. I'm wondering, then: why not leave that space to the user? We could, for instance, let the user define a baground image displayed on his stream page and visible for everyone on his profile page.

Flaburgan commented:
>Nope, tag page has too.

No, the grey used in tag page was the same than the left column. Previously in the PR, the bg grey used for the stream page was a darker one.

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -32,14 +32,19 @@
     display: block;
     font-weight: normal;
     padding: 10px 20px;
+    font-weight: bold;
```
Property `font-weight` already defined on line 37

Flaburgan commented:
I would love to see a review here, I want to continue my work on the design but I don't want this PR to become too big. Anyone wants to try it locally and tell me what he think about it?

Zauberstuhl commented:
@Flaburgan you tested it with the chat?

Flaburgan commented:
The chat is currently broken on d-fr I don't know why but yeah it integrates well :)

Flaburgan commented:
I discussed with @svbergerem about this new design, the main blocking point was the avatars of users in the stream in the middle of the left column. We both agree it doesn't fit well. So I need to either move it elsewhere (next to the stream title? at least in a place directly related with the stream) or to remove it (I heard about people saying it's useless). I created a poll in diaspora* to know what people think about that. Please answer and share this post: yourPodUrl.tld/posts/fbd4fa5098e7013319673e7c95cee2d9

Zauberstuhl commented:
I already wrote this on fla's post but I think that is important for others too

> I think Deus has a valid point there. Users are already able to switch themes.
> 
> For me for instance I restored the old design (stable) and added it to sechat as default.  
> But the  user is able to fall-back to the develop version anytime.
> 
> So create a new theme ..remove the user-avatars and let the podmin/user decide !

I mean it is called theme, but you can also customize the design, right?

kas70 commented:
@Flaburgan Can you post a screenshot with expanded chat-sidebar?

piratenpanda commented:
Are the single posts still separate boxes? I dont really like the strong separations. Why not leave the whole stream in a white block separated with light gray horizontal lines?

Flaburgan commented:
It's hard to rebase this on develop now that https://github.com/diaspora/diaspora/pull/6631 is merged. It looks easier to me to manually reapply the changes of @svbergerem on my branch. How should I say to git to ignore his commit after that?

Loelo commented:
Hi, I like Flaburgan's proposition. He suggested that the avatars of user in the stream could be placed next to the stream title, in this mockup here is what it would look like :
![mockup-up](https://cloud.githubusercontent.com/assets/11534708/12589010/dcf5e13a-c45b-11e5-8b36-1babf59252c3.png)

Flaburgan committed:
Fix original white theme

diaspora-code-review commented on app/views/aspects/_aspect_stream.haml:
```diff
@@ -3,15 +3,17 @@
-#gs-shim{:title => popover_with_close_html("3. #{t('.stay_updated')}"), 'data-content' => t('.stay_updated_explanation')}
-
-#main_stream.stream
+.stream#main_stream{title: popover_with_close_html("3. #{t('.stay_updated')}"),
```
Don't mix styles in the same hash.

diaspora-code-review commented on app/views/tags/show.haml:
```diff
@@ -33,8 +32,6 @@
-
-    .col-lg-1.visible-lg.right-sidebar-fixed-sub-background
-
+            
```
Line contains trailing whitespace

Flaburgan commented:
After an harmful rebase with a fix of the original white theme, I'd really want to hear from core members what to do to see this merged, it becomes harder everyday to keep it up-to-date with `develop`. I can assure that I'll continue to work on this to implement every remarks, I just don't want to add too many things to this because it starts to become impossible to review.

denschub commented:
Okay, I'll now give you more or less the same feedback as @svbergerem gave you in a private IRC session, but apparently, this information got lost somehow.

I see why you are trying to remove the right sidebar but I do not think the chat is really something that should drive this decision. Also, trying to unify all the different application views is really nice, but if you want to do it, you need to do it. I will absolutely focus on the look and feel and the generated source here, I have not looked at your actual source changes.

Status quo:

* The notification view is inside a `container-fluid`. The left menu is `col-md-3` without any background while the content is `col-md-9`, also without any background. Items in the navigation are bordered, links have a blue hover. Navigation font is `#555555`, not bold or anything.
* The conversation view is also inside a `container-fluid`, the left navigation is a `col-md-4` with a white background and a drop shadow. The content is a `col-md-8` with also a white background with a drop shadow. Navigation items are somewhat similar to the notifications.  However, the create view is in a `col-md-10` inside the `col-md-8`, so it's not actually full width. No idea why.
* User profiles are also inside a `container-fluid` and looks like the conversations view, but we have a `col-md-3`/`col-md-9` split here. The different split is fine since the left bar does not contain that much stuff. While looking on it at the moment, I don't really know why the `stream_container` there [has a `max-width` of `900px`](https://cloud.githubusercontent.com/assets/344777/12865272/0dfebd70-cca8-11e5-8c55-de9be12be919.png) since that looks rather weird. Otherwise, I'm quite happy with that view.
* The contacts view is a replica of the notification view. Whee!

Now, your proposed stream. First of all, I wonder why there is an empty `div`:

```html
<div class='left-navbar-fixed-background col-lg-2 col-sm-3 hidden-sm hidden-xs'></div>
```
That does not look like it is doing anything. Why's that?

So, now your split is 2/10, which is even smaller than a 3/9 split. When having [a first look on my usual resolution](https://cloud.githubusercontent.com/assets/344777/12865338/815c0eca-cca9-11e5-8c52-ee1d84158aca.png), it does not look too bad at all. However, [when looking at the stream in full-scren width](https://cloud.githubusercontent.com/assets/344777/12865354/b492e296-cca9-11e5-9cf9-f2cc440fdbbc.png), everything starts looking really bad. I am aware that the max-width is there to keep the lines readable, but the large space between the content and the navigation elements look horrible imho.

I hackily [added a `max-width` to the stream contents and the header](https://cloud.githubusercontent.com/assets/344777/12865440/d8ff26e2-ccab-11e5-8d63-a7eaeeb14b8e.png) and although this is a hack and it breaks stuff, it already looks better in my opinion. I'd rather have empty space outside the "important" area than all the UI elements scattered all around my display. Yes, that does apply to all the content views. However, it has to be done somewhere, so why not in this pull request.

Initially, I wanted to propose to simply keep the right bar. However, the more I look at it, the more I like it without the right bar. Our user interface is cluttered as hell and I feel like one panel less to look at would do no harm. Buuut, at the moment, there is way to much stuff going on in the left panel. So, some ideas to make it look less cramped:

* First of all, what do you (and the other people here) think about adopting the person/conversation-view for the stream? That is, moving the left panel contents into a box with drop shadows like the profile information on user profile pages. I think it would look nice. I'm talking about `.page-conversations .left-pane`s styling if unclear.
* We talked about this before and I really see no purpose of `.stream-faces`. If no one objects, kill it, kill it with fire.
* I really dislike the bold font. The navigation already draws enough attention without the bold items.
* I thought about making the "info boxes" expandable items. That is, make the headline clickable and expand the contents on click while keeping only one box open at a time.
    * It has to be clear that these items are not part of the navigation above. Maybe add a slight border between the groups and use a smaller font.
    * "Welcome new users" and "Need help?" could be merged into a "Need help or want to help?" box for example.
    * Some of the links in the footer could get optimized as well, but I think that's not important at the moment.

Opinions? Thoughts?

jaywink commented:
> We talked about this before and I really see no purpose of .stream-faces. If no one objects, kill it, kill it with fire.

Yes please, just get rid of it already :E

jaywink commented:
Btw, unrelated to comments regarding design. This is probably going to take a while and we've already got one UI refresh for 0.6 merged in. What about trying to ship 0.6 out asap (with some of the items in alpha status etc, like API, which ships only authentication) and then merge this in as soon as possible once design is ok to the devel branch for longer time testing. I'm afraid if we target this for 0.6 it will delay releasing that while inevitable tweaks and tunage happens. Plus skipping the design changes regarding 0.6 seems ... odd.

Flaburgan commented:
Thank you **very much** for your review. I'm not home this weekend but will have a look at it next week.

denschub commented:
#6675 added a global `max-width` of `1200px` (except for the stream, which has to be done here) and unifies the settings view to look like the notification/contacts view. We're getting somewhere! :)

svbergerem commented:
Let me go through [denschub's comment](https://github.com/diaspora/diaspora/pull/6535#issuecomment-180716500) and add some annotations:

> Status quo:
> [...]
> The conversation view is also inside a container-fluid [...]. However, the create view is in a col-md-10 inside the col-md-8, so it's not actually full width. No idea why.

This has been fixed by #6675.

> User profiles are also inside a container-fluid [...], I don't really know why the stream_container there has a max-width of 900px since that looks rather weird.

That is not the status quo but [something this PR added](https://github.com/Flaburgan/diaspora/commit/4f05e5e2c59b2a794776144a535969d03d5aaac4#diff-63cfe321ed66e43fe293b613037fa458R10). It's not needed anymore since we have the `max-width` for `.container-fluid`.

> I hackily [added a max-width to the stream contents and the header](https://cloud.githubusercontent.com/assets/344777/12865440/d8ff26e2-ccab-11e5-8d63-a7eaeeb14b8e.png) and although this is a hack and it breaks stuff, it already looks better in my opinion.

I agree.

+1 for everything else in the comment, especially

> moving the left panel contents into a box with drop shadows like the profile information on user profile pages

svbergerem commented:
I created a screenshot for a discussion and thought I'd also share it here and write down what I changed in case also others like some of the changes:

![8wcd00f](https://cloud.githubusercontent.com/assets/3798614/12995453/c60ddc38-d127-11e5-9e61-651c22e42e30.png)

### changes
* rebase
* remove `.left-navbar-fixed-background`
* add `$card-shadow` as box-shadow to `.left-navbar`
* remove negative margins from `#stream_selection`
* add padding to `.info-bar` (left and right 20px)
* remove max-width from `.stream_container` and remove `max-width: 100%` for the `.container-fluid` on stream pages
* remove `.col-lg-2` and `.col-lg-10`, only keep `.col-md-3` and `.col-md-9`
* move `.left-navbar` into of the `.col-md-3`
* remove haml code for stream faces (somehow that was still left after the rebase)

@Flaburgan I can send you a diff if you are interested but you can also just use this list and the screenshot as some additional input for your own ideas.

jaywink commented:
I must say the fact that the left bar is not locked to the side but "with" the content looks much better - that was the only thing I felt out of place with Fla's proposal. Great work guys :+1:

Flaburgan committed:
Max-width for the whole content

Flaburgan commented:
I updated this PR, it now looks like Steffen screenshot. It's also running on d-fr so I hope I'll be able to find the last bugs in the next few days. The CSS code could probably be more cleaned too, but the design should be in its final state now.

svbergerem commented:
Here are the open issues from @denschub s comment:

> I really dislike the bold font. The navigation already draws enough attention without the bold items.

I haven't decided yet, which version I'd prefer.

> I thought about making the "info boxes" expandable items. That is, make the headline clickable and expand the contents on click while keeping only one box open at a time.
> * It has to be clear that these items are not part of the navigation above. Maybe add a slight border between the groups and use a smaller font.
> * "Welcome new users" and "Need help?" could be merged into a "Need help or want to help?" box for example.
> * Some of the links in the footer could get optimized as well, but I think that's not important at the moment.

Here is a diff that implements some basics:
https://gist.github.com/svbergerem/1755e57a487889804f8e

And this is how it looks like:
![info-bar expandable](https://cloud.githubusercontent.com/assets/3798614/13207117/f72cf7a0-d90a-11e5-96be-c274efd52e48.png)

I did no polishing and I also didn't work on all points mentioned above but I thought this could help to finish this PR.

Flaburgan committed:
collapse left navbar sections

Flaburgan commented:
PR updated with your changes. I like it, the nav is lighter that way. I kept the links not collapsed though, it felt better.

Also, I noticed a regression, if you mouse over anywhere in the stream you have an alt text "3. stay updated...". This comes from the getting started info window and the regression is caused by [this change](https://github.com/diaspora/diaspora/pull/6535/files#diff-053916cb96b2b86565847c1079098286L119) but I don't get why we have that text anyway, especially when getting started hints are not activated...

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -136,3 +99,79 @@
+  }
+
+  .section:not(.collapsed) .entypo-triangle-right { display: none }
+  .section.collapsed {
```
Merge rule `.section.collapsed` with rule on line 106

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -136,3 +99,79 @@
+
+  .title {
+    border-bottom: 1px solid $border-grey;
+    padding-bottom: 5px;
```
Properties should be ordered border-bottom, cursor, padding-bottom

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -136,3 +99,79 @@
+    margin-top: 20px;
+  }
+
+  .section:not(.collapsed) .entypo-triangle-right { display: none }
```
Declaration should be terminated by a semicolon

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -136,3 +99,79 @@
+
+  .section:not(.collapsed) .entypo-triangle-right { display: none }
+  .section.collapsed {
+    .entypo-triangle-down { display: none}
```
Declaration should be terminated by a semicolon

svbergerem commented on app/views/aspects/_aspect_stream.haml:
```diff
@@ -3,14 +3,16 @@
-
-#main_stream.stream
+.stream#main_stream{:title => popover_with_close_html("3. #{t('.stay_updated')}"),
+  "data-content" => t(".stay_updated_explanation")}
```
> I don't get why we have that text anyway, especially when getting started hints are not activated...

I think we can check `current_user.getting_started?` before adding the title and `data-content`.

Flaburgan replied:
Is there a syntax to factorize `.stream#main_stream` or do I have to write
```
- if ...
  .stream#main_stream{...}
- else
  .stream#main_stream

svbergerem replied:
There might be nicer solutions but at least this one is working:
```diff
diff --git a/app/views/aspects/_aspect_stream.haml b/app/views/aspects/_aspect_stream.haml
index abee4ac..7dcca6a 100644
--- a/app/views/aspects/_aspect_stream.haml
+++ b/app/views/aspects/_aspect_stream.haml
@@ -11,8 +11,8 @@
     = owner_image_link
   = render "publisher/publisher", publisher_aspects_for(stream)
 
-.stream#main_stream{:title => popover_with_close_html("3. #{t('.stay_updated')}"),
-  "data-content" => t(".stay_updated_explanation")}
+.stream#main_stream{:title => current_user.getting_started? ? popover_with_close_html("3. #{t('.stay_updated')}") : "",
+  "data-content" => current_user.getting_started? ? t(".stay_updated_explanation") : ""}
 
 #paginate
   %span.loader.hidden
```

Flaburgan replied:
Hm, that means two empty attributes, not really clean. I'm going to stay with the if / else here I think.

svbergerem commented on app/views/streams/main_stream.html.haml:
```diff
@@ -20,38 +20,144 @@
           .pull-right
             = link_to '&times;'.html_safe, getting_started_completed_path, id: "gs-skip-x", class: "close"
```
This block looks like it could need some love. I'd say we could use `.container-fluid` instead of `.container`, remove that `.offset1` which wasn't working at all and instead use `.col-md-9` and `.col-md-3` or something similar.

svbergerem commented:
The headline in the sidebar for the tag stream looks too big. It has more text than other headlines (e.g. “1 person tagged with #diaspora”) so I'd prefer a smaller headline there.

I also think that the sidebar for the main stream still needs some improvements. To me it looks like there is some separation missing between the stream selection and the info-bar but I don't have any good idea right now how to do that. Reducing the space between two bullet points in the info-bar could also help. `margin-top: 10px` instead of 20px for `.info-bar .section` looks good to me.

diaspora-code-review commented on app/assets/stylesheets/navbar_left.scss:
```diff
@@ -136,3 +99,87 @@
+      display: none;
+    }
+
+    &.collapsed .title h5 {
```
Selector should have depth of applicability no greater than 3, but was 4

diaspora-code-review commented on app/assets/stylesheets/tag.scss:
```diff
@@ -23,11 +23,13 @@ h1.tag {
-    .col-md-3 {
-      h4 { margin: 25px 0 15px; }
+    .sidebar {
+      h3 {
```
Nesting should be no greater than 3, but was 4

diaspora-code-review replied:
Selector should have depth of applicability no greater than 3, but was 4

Flaburgan replied:
I'll remove the `.page-tags` to reduce the depth in a following PR: it modifies every lines so pronto is crying about ids used etc, I don't want to do any more noise here.

Flaburgan commented:
PR updated:
- Fixed tag title size
- Fixed white background on mobile
- Refactor of Getting Started orange panel
- Add a border to separate streams selection, sections and links. Section titles don't have a border-bottom anymore but become bold when opened, see below:

![capture du 2016-03-08 09-53-46](https://cloud.githubusercontent.com/assets/930064/13598579/dd7613bc-e51e-11e5-92f3-eeacbdca9284.png)

Flaburgan committed:
Fix tests

diaspora-code-review commented on app/assets/stylesheets/single-post-view.scss:
```diff
@@ -4,8 +4,7 @@
   .head {
-    padding-bottom: 10px;
-    padding-top: 5px;
+    padding: 10px 0;
```
Properties should be ordered border-bottom, padding

diaspora-code-review commented on app/assets/stylesheets/single-post-view.scss:
```diff
@@ -108,6 +107,11 @@
+  & > .framed-content {
+    padding-top: 10px;
+  }
+  
```
Line contains trailing whitespace

diaspora-code-review commented on app/assets/stylesheets/single-post-view.scss:
```diff
@@ -108,6 +107,11 @@
 
 #single-post-interactions {
+
+  & > .framed-content {
```
Unnecessary parent selector (&)

svbergerem commented:
@Flaburgan Here is what I found together with @denschub and @SuperTux88 for the separation:

![sidebar separation](https://cloud.githubusercontent.com/assets/3798614/13799043/434a1a64-eb1c-11e5-9101-5056febe8b7f.png)

```diff
diff --git a/app/assets/stylesheets/navbar_left.scss b/app/assets/stylesheets/navbar_left.scss
index 51adb58..df6df71 100644
--- a/app/assets/stylesheets/navbar_left.scss
+++ b/app/assets/stylesheets/navbar_left.scss
@@ -1,5 +1,6 @@
 .left-navbar {
   padding: 0;
+  padding-bottom: 10px;
 
   a {
     text-decoration: none;
@@ -101,8 +102,14 @@
 }
 
 .info-bar {
-  border-top: 1px solid $border-grey;
-  padding: 0 20px;
+  margin-top: 20px;
+  padding: 10px 20px 0;
+
+  .info-links {
+    border-top: 1px solid $border-grey;
+    margin-top: 10px;
+    padding: 5px;
+  }
 
   .section {
     margin-top: 10px;
diff --git a/app/views/streams/main_stream.html.haml b/app/views/streams/main_stream.html.haml
index 51fdc5c..25c4155 100644
--- a/app/views/streams/main_stream.html.haml
+++ b/app/views/streams/main_stream.html.haml
@@ -38,118 +38,118 @@
           %li{data: {stream: "public"}}
             = link_to t("streams.public.title"), public_stream_path, rel: "backbone", class: "hoverable"
 
-        .info-bar.hidden-xs
-          - if AppConfig.settings.invitations.open?
-            .section.collapsed
-              .title
-                %h5.title-header
-                  .entypo-triangle-right
-                  .entypo-triangle-down
-                  = t("shared.invitations.invite_your_friends")
-              .content
-                = render "shared/invitations"
-
+      .sidebar.info-bar.hidden-xs
+        - if AppConfig.settings.invitations.open?
           .section.collapsed
             .title
               %h5.title-header
                 .entypo-triangle-right
                 .entypo-triangle-down
-                = t("aspects.index.new_here.title")
+                = t("shared.invitations.invite_your_friends")
             .content
-              != t("aspects.index.new_here.follow",
-                link: link_to("#" + t("shared.publisher.new_user_prefill.newhere"),
-                tag_path(name: t("shared.publisher.new_user_prefill.newhere"))))
-              %br
-              = link_to(t("aspects.index.new_here.learn_more"),
-                "http://wiki.diasporafoundation.org/Welcoming_Committee")
+              = render "shared/invitations"
+
+        .section.collapsed
+          .title
+            %h5.title-header
+              .entypo-triangle-right
+              .entypo-triangle-down
+              = t("aspects.index.new_here.title")
+          .content
+            != t("aspects.index.new_here.follow",
+              link: link_to("#" + t("shared.publisher.new_user_prefill.newhere"),
+              tag_path(name: t("shared.publisher.new_user_prefill.newhere"))))
+            %br
+            = link_to(t("aspects.index.new_here.learn_more"),
+              "http://wiki.diasporafoundation.org/Welcoming_Committee")
+
+        .section.collapsed
+          .title
+            %h5.title-header
+              .entypo-triangle-right
+              .entypo-triangle-down
+              = t("aspects.index.help.need_help")
+          .content
+            %p
+              = t("aspects.index.help.here_to_help")
+            %p
+              = t("aspects.index.help.do_you")
+            %ul
+              %li
+                != t("aspects.index.help.have_a_question",
+                  link: link_to("#" + t("aspects.index.help.tag_question"),
+                  tag_path(name: t("aspects.index.help.tag_question"))))
+              %li
+                != t("aspects.index.help.find_a_bug",
+                  link: link_to("#" + t("aspects.index.help.tag_bug"),
+                  tag_path(name: t("aspects.index.help.tag_bug"))))
+              %li
+                != t("aspects.index.help.feature_suggestion",
+                  link: link_to("#" + t("aspects.index.help.tag_feature"),
+                  tag_path(name: t("aspects.index.help.tag_feature"))))
+            %p
+              != t("aspects.index.help.tutorials_and_wiki",
+                faq: link_to(t("_help"), help_path),
+                tutorial: link_to(t("aspects.index.help.tutorial_link_text"),
+                  "https://diasporafoundation.org/tutorials", target: "_blank"),
+                wiki: link_to("Wiki", "http://wiki.diasporafoundation.org",
+                  target: "_blank"),
+                target: "_blank")
 
+        - unless AppConfig.configured_services.blank? || all_services_connected?
           .section.collapsed
             .title
               %h5.title-header
                 .entypo-triangle-right
                 .entypo-triangle-down
-                = t("aspects.index.help.need_help")
+                = t("aspects.index.services.heading")
             .content
-              %p
-                = t("aspects.index.help.here_to_help")
-              %p
-                = t("aspects.index.help.do_you")
-              %ul
-                %li
-                  != t("aspects.index.help.have_a_question",
-                    link: link_to("#" + t("aspects.index.help.tag_question"),
-                    tag_path(name: t("aspects.index.help.tag_question"))))
-                %li
-                  != t("aspects.index.help.find_a_bug",
-                    link: link_to("#" + t("aspects.index.help.tag_bug"),
-                    tag_path(name: t("aspects.index.help.tag_bug"))))
-                %li
-                  != t("aspects.index.help.feature_suggestion",
-                    link: link_to("#" + t("aspects.index.help.tag_feature"),
-                    tag_path(name: t("aspects.index.help.tag_feature"))))
-              %p
-                != t("aspects.index.help.tutorials_and_wiki",
-                  faq: link_to(t("_help"), help_path),
-                  tutorial: link_to(t("aspects.index.help.tutorial_link_text"),
-                    "https://diasporafoundation.org/tutorials", target: "_blank"),
-                  wiki: link_to("Wiki", "http://wiki.diasporafoundation.org",
-                    target: "_blank"),
-                  target: "_blank")
+              %div
+                = t("aspects.index.services.content")
 
-          - unless AppConfig.configured_services.blank? || all_services_connected?
-            .section.collapsed
-              .title
-                %h5.title-header
-                  .entypo-triangle-right
-                  .entypo-triangle-down
-                  = t("aspects.index.services.heading")
-              .content
-                %div
-                  = t("aspects.index.services.content")
+              .right-service-icons
+                - AppConfig.configured_services.each do |service|
+                  - if AppConfig.show_service?(service, current_user)
+                    - unless current_user.services.any? {|x| x.provider == service }
+                      = link_to(content_tag(:div, nil,
+                        class: "social-media-logos-#{service.to_s.downcase}-24x24",
+                        title: service.to_s.titleize), "/auth/#{service}")
 
-                .right-service-icons
-                  - AppConfig.configured_services.each do |service|
-                    - if AppConfig.show_service?(service, current_user)
-                      - unless current_user.services.any? {|x| x.provider == service }
-                        = link_to(content_tag(:div, nil,
-                          class: "social-media-logos-#{service.to_s.downcase}-24x24",
-                          title: service.to_s.titleize), "/auth/#{service}")
+        .section.collapsed
+          .title
+            %h5.title-header
+              .entypo-triangle-right
+              .entypo-triangle-down
+              = t("bookmarklet.heading")
+          .content
+            != t("bookmarklet.explanation", link: link_to(t("bookmarklet.post_something"), bookmarklet_code))
 
+        - if AppConfig.settings.paypal_donations.enable? || AppConfig.bitcoin_donation_address
           .section.collapsed
             .title
               %h5.title-header
                 .entypo-triangle-right
                 .entypo-triangle-down
-                = t("bookmarklet.heading")
+                = t("aspects.index.donate")
             .content
-              != t("bookmarklet.explanation", link: link_to(t("bookmarklet.post_something"), bookmarklet_code))
-
-          - if AppConfig.settings.paypal_donations.enable? || AppConfig.bitcoin_donation_address
-            .section.collapsed
-              .title
-                %h5.title-header
-                  .entypo-triangle-right
-                  .entypo-triangle-down
-                  = t("aspects.index.donate")
-              .content
-                %p
-                  = t("aspects.index.keep_pod_running", pod: AppConfig.pod_uri.host)
-                = render "shared/donatepod"
+              %p
+                = t("aspects.index.keep_pod_running", pod: AppConfig.pod_uri.host)
+              = render "shared/donatepod"
 
-          - if AppConfig.admins.podmin_email.present?
-            .section.collapsed
-              .title
-                %h5.title-header
-                  .entypo-triangle-right
-                  .entypo-triangle-down
-                  = t("aspects.index.help.any_problem")
-              .content
-                %p
-                  = t("aspects.index.help.contact_podmin")
-                %p
-                  = link_to t("aspects.index.help.mail_podmin"), "mailto:#{AppConfig.admins.podmin_email}"
+        - if AppConfig.admins.podmin_email.present?
+          .section.collapsed
+            .title
+              %h5.title-header
+                .entypo-triangle-right
+                .entypo-triangle-down
+                = t("aspects.index.help.any_problem")
+            .content
+              %p
+                = t("aspects.index.help.contact_podmin")
+              %p
+                = link_to t("aspects.index.help.mail_podmin"), "mailto:#{AppConfig.admins.podmin_email}"
 
-        .info-bar
+        .info-links
           .content
             %ul
               = render "shared/links"
```

And here is a screenshot of the tag stream that I mentioned earlier:
![tag stream sidebar](https://cloud.githubusercontent.com/assets/3798614/13799044/434cab76-eb1c-11e5-8eff-527932f72293.png)

Some margin (bottom and top) might help for the sidebar content on the tag stream page.

It would be great if you could also fix the code style remarks. Then I'll do another review and see if I can find any bugs.

I'm really looking forward to merging this PR. :)

svbergerem commented:
The tests are currently failing because the invitations button is not visible. We have to click on the section header first.

Flaburgan commented:
>The tests are currently failing because the invitations button is not visible. We have to click on the section header first.

Yeah I noticed that, and also the code style, I just didn't have time to work on it :p 

I'm going to apply your patch and fix everything, hopefully before the end of the week-end, I'm kind of sick actually :(

Flaburgan committed:
split left column

Flaburgan commented on app/assets/stylesheets/stream.scss:
```diff
@@ -1,6 +1,25 @@
+  }
+}
+
+@media(max-width: $screen-xs) {
```
Looks like this is not sync with `.hidden-xs`. Anybody knows why?

svbergerem replied:
Maybe try `$screen-xs-max`.

Flaburgan replied:
It works :) Waiting for the build to complete then I'll commit and you'll be able to do the final review.

Flaburgan commented:
Tests are green :)

svbergerem commented:
The code looks good to me. I found a little regression but that was small enough that I just fixed it myself by adding `.clearfix` to the conversations participants. Before there was a gap between the participants and the header. Apart from that everything looked fine.

Thanks a lot. The new stream design looks awesome. :heart:

Flaburgan commented:
Thank you! I'm really happy to see this finally integrated :)
